lib/jwt/decode.rb

```diff
@@ -13,7 +13,7 @@ def initialize(jwt, key, verify, options, &keyfinder)
       @jwt = jwt
       @key = key
       @options = options
-      @segments = jwt.split('.')
+      @segments = jwt.split('.', -1)
```

Comment:
`#split` by default ignores empty strings. This change is needed to correctly identify an empty string signature in a JWT structured according to https://tools.ietf.org/html/rfc7519#section-7.2 and https://tools.ietf.org/html/rfc7516#section-9